sharding/indexHashedNodesCoordinator.go

```diff
@@ -496,15 +496,18 @@ func (ihgs *indexHashedNodesCoordinator) EpochStartPrepare(metaHdr data.HeaderHa
 	}
 
 	shufflerArgs := ArgsUpdateNodes{
-		Eligible: newNodesConfig.eligibleMap,
-		Waiting:  newNodesConfig.waitingMap,
+		Eligible: toOrderedSlice(newNodesConfig.eligibleMap, newNodesConfig.nbShards),
+		Waiting:  toOrderedSlice(newNodesConfig.waitingMap, newNodesConfig.nbShards),
 		NewNodes: newNodesConfig.newList,
 		Leaving:  newNodesConfig.leavingList,
 		Rand:     randomness,
 		NbShards: newNodesConfig.nbShards,
 	}
 
-	eligibleMap, waitingMap, leaving := ihgs.shuffler.UpdateNodeLists(shufflerArgs)
+	eligibleLists, waitingLists, leaving := ihgs.shuffler.UpdateNodeLists(shufflerArgs)
```

Comment:
Why does the UpdateNodeLists update a slice of slices? Why not a map directly?

Reply:
undone. fixed